src/ast/passes/codegen_llvm.cpp

```diff
@@ -1154,7 +1114,243 @@ ScopedExpr CodegenLLVM::visit(Call &call)
         map, scoped_key.value(), b_.getInt64(1), call.loc);
 
     return ScopedExpr();
+  } else if (call.func == "tseries") {
+    // tseries decides what the current bucket is based on the timestamp then
+    // updates the bucket's value.
+    //
+    // void tseries(uint64_t value, uint64_t interval_ns, uint64_t n) {
+    //   uint64_t now = bpf_ktime_get_boot_ns();
+    //   struct ts_struct ts_struct_alloc = {};
+    //   uint64_t epoch = now / interval_ns;
+    //   uint64_t bucket = epoch % n;
+    //   struct ts_struct *bucket_value;
+    //   bool key_exists;
+    //
+    //   bucket_value = bpf_map_lookup_elem(&tseries_map, &bucket);
+    //   if (!bucket_value) {
+    //     key_exists = false;
+    //     bucket_value = &ts_struct_alloc;
+    //   } else {
+    //     key_exists = true;
+    //   }
+    //
+    // #if defined(SUM) || defined(MIN) || defined(MAX) || defined(AVG)
```

Comment:
Nit: these aren't really ifdefs but it's pseudo code so I won't block. Also, this comment is very helpful. Thanks!